keel-artifact/src/main/kotlin/com/netflix/spinnaker/keel/artifacts/ArtifactListener.kt

```diff
@@ -129,16 +129,16 @@ class ArtifactListener(
             if (latestArtifact != null) {
               val hasNew = when {
                 lastRecordedVersion == null -> true
-                latestArtifact.version != lastRecordedVersion -> {
-                  artifact.versioningStrategy.comparator.compare(lastRecordedVersion, latestArtifact.version) > 0
+                latestArtifact.version != lastRecordedVersion.version -> {
```

Comment:
I know. I struggled with these names a bit because the table is called `artifact_versions` and the methods are called `somethingArtifactVersion`, while the class is currently named `PublishedArtifact`, and we generally use "artifact" to refer to a `DeliveryArtifact`... 😓